### ISSUE TITLE
Dependencies during installation of pyroSAR

### ISSUE THREAD

**Issue by sid-ks** [johntruckenbrodt/pyroSAR#140]
Hi, just a quick question.

Why does pyroSAR install matplotlib, jupyter, notebook and other such dependencies when doing a pip3 install?

Is there a way to make it more lean, with only pure dependencies of pyroSAR being installed?

If so, can you tell me how, if not, please explain why these are required.

Im quite a fan of pyroSAR and how actively it is maintained and updated. Thanks for your efforts! :)

**Comment by sid-ks** [comment]
> Hi, just a quick question.
> 
> Why does pyroSAR install matplotlib, jupyter, notebook and other such dependencies when doing a pip3 install?
> 
> Is there a way to make it more lean, with only pure dependencies of pyroSAR being installed?
> 
> If so, can you tell me how, if not, please explain why these are required.
> 
> Im quite a fan of pyroSAR and how actively it is maintained and updated. Thanks for your efforts! :)

Any comments on this please? Thank you. :)

**Comment by johntruckenbrodt** [comment]
Hi @sid-ks, thanks for addressing this issue. `pyroSAR` depends on [spatialist](https://github.com/johntruckenbrodt/spatialist) for general spatial data handling and several other convenience functionalities. This package has grown to be quite a large toolbox over the years and some functionality is demonstrated in a Jupyter notebook [spatialist_tutorial.ipynb](https://github.com/johntruckenbrodt/spatialist/blob/master/spatialist_tutorial.ipynb). The dependencies are mainly needed for a visualization class [RasterViewer](https://spatialist.readthedocs.io/en/latest/spatialist.html#spatialist.explorer.RasterViewer), which is part of `spatialist` and whose use is also demonstrated in the notebook.  
So basically, these dependencies are not directly needed by `pyroSAR`. However, excluding them would mean I would have to split up `spatialist` to outsource the visualization components. I am developing `pyroSAR` and `spatialist` almost alone and the more packages I need to maintain the more work needs to be done, which unfortunately I cannot justify at the moment.  
I agree though that making the package more lean is important. I will leave this open and try to find time for it.

**Comment by sid-ks** [comment]
Wow, really appreciate your time and efforts. I would be very glad to help out in someway, but its quite hectic at work even for me, and also being a beginner I wouldn't wanna slow down your schedule.
Thanks for keeping it open, and I look forward to the time when this can be implemented. 

Great package ! Have a nice day! :)

**Comment by johntruckenbrodt** [comment]
Thanks mate! I had a quick search of how to solve this with little effort. With pip you have the [PEP 508](https://www.python.org/dev/peps/pep-0508/#extras) convention. For the conda package the best approach seems to be to split `spatialist` into `spatialist-base` (core functionality) and `spatialist` (meta package with optional packages). This is for example done by `matplotlib`.  
If you like you could modify the requirements files of `spatialist` and send me a pull request. There are two, but I think they could also be combined into one using the PEP 508 convention. For example by adding two sections `testing` and `visualization`. Once this is done I can set up the conda packages.
